tuplex/core/include/physical/PhysicalStage.h

```diff
@@ -44,16 +48,23 @@ namespace tuplex {
         std::vector<PhysicalStage*> _predecessors;
         int64_t _number;
         std::unordered_map<std::tuple<int64_t, ExceptionCode>, size_t> _ecounts; //! exception counts for this stage.
+        std::vector<LogicalOperator*> _opids;
     protected:
         IBackend* _backend;
+        std::shared_ptr<HistoryServerConnector> _historyServer;
     public:
+        void setHistoryServer(std::shared_ptr<HistoryServerConnector> hsc) { _historyServer = hsc; }
```

Comment:
is this multi-threading safe?

Reply:
I would say yes. This method is only called in LocalBackend::execute. And to my knowledge, the stages are executed in a linear fashion.